### PR TITLE
Add include instructions processing

### PR DESCRIPTION
**Pull request by frolui**

### CODE REVIEW

**Comment by frolui** [comment]
results without include 
image

**Comment by frolui** [comment]
results with include
image

File: make_profiler/__main__.py [hunk 1]
```diff
@@ -56,7 +56,7 @@ def main(argv=sys.argv[1:]):
 
     args, unknown_args = parser.parse_known_args(argv)
 
-    in_file = open(args.in_filename, 'r')
+    in_file = check_include_instruction(args.in_filename)
```
Comment (Komzpa) [inline make_profiler/__main__.py:59]: Does not look good here, "file = check" does not match my expectations (I expect check to return boolean)

File: make_profiler/parser.py [hunk 2]
```diff
@@ -127,3 +129,47 @@ def recurse_indirect_influences(original_target, recurse_target):
             recurse_indirect_influences(original_target, t)
 
     return dependencies, influences, order_only, indirect_influences
+
+
+def check_include_instruction(filename):
```
Comment (Komzpa) [inline make_profiler/parser.py:134]: I would expect this to be a part of parse() function.

**Review by Komzpa — CHANGES_REQUESTED** [review]

File: make_profiler/parser.py [hunk 3]
```diff
@@ -127,3 +129,47 @@ def recurse_indirect_influences(original_target, recurse_target):
...
+    matches = [re.sub(' +', ' ', string) for string in lines if re.match(regex, string)]
+    
+    # check if input make contains include instructions
+    if len(matches) == 0:
+        temp_make_file = open(filename, 'r')
+        
+    else:
+        # create list of included makes
+        # we use join by space and then split by space to process multiple included instructions
+        makes = ' '.join([x.split('include ')[1] for x in matches]).split(' ')
```
Comment (Komzpa) [inline make_profiler/parser.py:153]: I believe that the order of includes is significant and includes need to happen at the spot they were mentioned.

**Comment by frolui** [comment]
rewrited after review
image

File: make_profiler/__main__.py (hunk 1 above)
Comment (frolui) [reply make_profiler/__main__.py]: renamed to **_insert_included_files_**

File: make_profiler/parser.py (hunk 2 above)
Comment (frolui) [reply make_profiler/parser.py]: moved inside parse() function

File: make_profiler/parser.py (hunk 3 above)
Comment (frolui) [reply make_profiler/parser.py]: changed to write exactly on place of instruction with keeping order of files in instruction

**Pull request merged**